recOrder/preproc/pre_processing.py

```diff
@@ -45,3 +45,34 @@ def brenner_gradient(im):
     focus_idx_max = np.where(focus_scores == np.max(focus_scores))[0][0]
 
     return focus_idx_max, focus_idx_min
+
+def get_autocontrast_limits(img, clip = .01):
+
+    #TODO: Figure out how to do this for float images (ret, phase, etc.)
+    # current only works for uint8, uint16, etc.
+
+    data_type = img.dtype
+
+    # Will raise an error if dtype is float, in that case cap the n_bins at 65536
+    try:
+        n_bins = int(np.iinfo(data_type).max)
+    except:
+        n_bins = 65536
+
+    hist = cv2.calcHist([img.astype(data_type.name)], [0], None, [n_bins], [0, n_bins])
+    cdf = np.cumsum(hist)
+
+    maximum = cdf[-1]
+    pix_clip = maximum * clip
+
```

Comment:
otsu and rosin are histogram-threshold based methods that accomplish a similar thing.  You might want to consider if cv2 has these implemented

Reply:
I think this method is typically used by imagej, MM, and other programs to specify a percentage of the high/low pixels to ignore.  There are some other fancier methods I looked into like CLAHE but seemed unnecessary to implement for our case of display.